data_structures/src/Stack/DynamicStack.cs

```diff
@@ -61,6 +67,18 @@ private void Resize()
         {
             //This function needs to resize the items array when called.
             //Calling this function will result in the items array growing or shrinking
+            if(Capacity == MIN_SIZE || _top % MIN_SIZE > 2)
+            {
+                var copy = new T[Capacity + MIN_SIZE];
+                _items.CopyTo(copy, 0);
+                _items = copy;
+            }        
+            else
+            {
+                var copy = new T[Capacity - MIN_SIZE];
```

Comment:
The way you are shrinking your array will cause [thrashing](https://en.wikipedia.org/wiki/Thrashing_(computer_science)).